user/views.py

```diff
@@ -61,12 +64,15 @@ def signup(request):
             if models.User.objects.filter(email=email).first() is not None:
                 messages.error(request, 'An account with this email already exists')
             else:
-                user = models.User.objects.create_user(email=email, password=password, name=name)
+                user = models.User.objects.create_user(email=email, password=password, name=name, u_type=u_type)
                 user = auth.authenticate(email=email, password=password)
                 auth.login(request, user)
                 return HttpResponseRedirect(reverse('root'))
     else:
-        form = forms.RegisterForm()
+        if u_type == models.USR_URL_SPONSOR:
+            form = forms.RegisterSponsorForm()
+        else:
+            form = forms.RegisterForm()
```

Comment:
Probably store a dict for given a u_type what Form instance they should use.

Reply:
I would have to create uno for the () and another for the post. I think an if it's better